compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -946,6 +946,13 @@ fn generic_simd_intrinsic<'ll, 'tcx>(
         tcx.normalize_erasing_late_bound_regions(ty::ParamEnv::reveal_all(), callee_ty.fn_sig(tcx));
     let arg_tys = sig.inputs();
 
+    // Vectors must be immediates (non-power-of-2 #[repr(packed)] are not)
+    for (ty, arg) in arg_tys.iter().zip(args) {
+        if ty.is_simd() && !matches!(arg.val, OperandValue::Immediate(_)) {
+            return_error!(InvalidMonomorphization::SimdArgument { span, name, ty: *ty });
+        }
+    }
```

Comment:
This comment seems no longer accurate... maybe that got changed by https://github.com/rust-lang/rust/pull/125311? Doing simd_mul etc on a packed SIMD type works just fine now. Even in debug builds this [generates the code one would hope for](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=49bba1357ef904726ba123e3afbc3545).